File: python/process.py
from transform import transform_data


def process_data(json_data):
    """
    Extracts 'value' fields from each record. Assumes every record has a 'attributes' and 'value'.
    Corner cases:
    - If a record does not have an 'attributes' or 'value' field, the function throws KeyError.
    """
    extracted_data = [record["attributes"]["value"] for record in json_data]
    return transform_data(extracted_data)

File: typescript/src/process.ts
import { Tranform } from "./transform";


export function processData(jsonData: any[]) {
  // Extracts 'value' fields from each record.
  const extractedData = jsonData.map(record => {
    if (!record.attributes || record.attributes.value === undefined) {
      throw new Error('Missing attributes or value field');
    }
    return record.attributes.value;
  });
  return Tranform.transformData(extractedData);
}
